### PR TITLE
implement edit-transaction and refactor some files

### PR DESCRIPTION
**Pull request by p-enz**
implement edit- transaction function in useStore
we reused our add transaction form for the edit
refactor our db-file that it matches our requirements
add new routes in app.js

### CODE REVIEW

File: src/components/AddTransaction/index.js [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
 import useStore from '../../hooks/useStore';
 import {Form} from '../AddNewAccount/styled';
 
-export default function AddTranscation() {
+export default function AddTransaction() {
 	const [transaction, setTransaction] = useState({
 		date: '',
 		change: '',
 		note: '',
 		category: '',
```
Comment (macfrei) [inline src/components/AddTransaction/index.js:12]: Maybe add a new variable "initialTransaction" which refers to your initial state. This way you can use the variable later as well and won't repeat yourself. Just a nice and quick way to reset a state 🙂

File: src/components/AddTransaction/index.js [hunk 2]
```diff
@@ -1,26 +1,50 @@
...
 				onSubmit={event => {
 					event.preventDefault();
-					addTransaction(accountID, transaction);
+					if (transactionID) {
+						editTransaction(accountID, transaction);
+						alert('edited');
+					} else {
+						addTransaction(accountID, transaction);
+					}
 					setTransaction({date: '', change: '', note: '', category: ''});
```
Comment (macfrei) [inline src/components/AddTransaction/index.js:48]: Just a personal preference, however, I would write and call a function `handleSubmit` which does everything you do here. It'd just be a bit out of sight 😉

**Review by stefankorder — APPROVED** [review]
Good work! 👍🏻

File: src/components/AddTransaction/index.js [hunk 3]
```diff
@@ -1,26 +1,50 @@
...
 
 	return (
 		<>
 			<Form
 				onSubmit={event => {
 					event.preventDefault();
-					addTransaction(accountID, transaction);
+					if (transactionID) {
+						editTransaction(accountID, transaction);
+						alert('edited');
```
Comment (macfrei) [inline src/components/AddTransaction/index.js:44]: I would probably not work with an `alert`. If you want notifications, you could have a look at [React Toastify](https://www.npmjs.com/package/react-toastify)

File: src/components/Transaction/Transactions.js [hunk 4]
```diff
@@ -9,11 +10,20 @@ export default function Transactions() {
...
 		<>
 			{entries.map(({id, date, change, note}) => (
-				<Transaction key={id} date={date} change={change} note={note} />
+				<Transaction
+					key={id}
+					date={date}
+					change={change}
+					note={note}
+					onEdit={() => {
+						navigate('/transaction/edit/' + accountID + '/' + id);
```
Comment (macfrei) [inline src/components/Transaction/Transactions.js:24]: I usually prefer template strings but both are valid solutions :)
```suggestion
						navigate(`/transaction/edit/${accountID}/${id}`);
```

**Review by macfrei — COMMENTED** [review]
Nice job 👍 Just added a few comments, mostly other ways to do the same thing 😉

File: src/components/AddTransaction/index.js (hunk 2 above)
Comment (dananhsl) [reply src/components/AddTransaction/index.js:48]: Thanks!! We will keep that in mind once we refactor our code!

File: src/components/AddTransaction/index.js (hunk 3 above)
Comment (dananhsl) [reply src/components/AddTransaction/index.js:44]: We thought so too and we want to implement toastify in a seperate user story to make alerts happen whenever the user edits, adds or deletes something!

File: src/components/AddTransaction/index.js (hunk 1 above)
Comment (stefankorder) [reply src/components/AddTransaction/index.js:12]: For example you can use the variable in the set Transaction function at line 48 at the index.js.

Comment (dananhsl) [reply src/components/AddTransaction/index.js:12]: Oh! Yeah that makes total sense.. 😂 My thoughts were way more complicated than what it actually is. Keep it simple! Thank you.

Comment (stefankorder) [reply src/components/AddTransaction/index.js:12]: 👍🏻😃

**Pull request merged**